src/server/DeviceServer_v2.js

```diff
@@ -181,6 +104,94 @@ class DeviceServer {
     );
   }
 
+  _onNewSocketConnection = async (socket: Socket): Promise<void> => {
+    try {
+      // eslint-disable-next-line no-plusplus
+      const connectionKey = `_${connectionIdCounter++}`;
+      const device = new SparkCore(socket, connectionKey);
+
+      device.on(
+        DEVICE_EVENT_NAMES.READY,
+        (): void => this._onDeviceReady(device),
```

Comment:
not sure if I should or shouldn't pass all subscriptions except `ready` and `disconnect`  into this._onDeviceReady. It works now as it is, but logically, all others subscriptions should be after `handshake` , because they use inner device data like `deviceID`